mydemoproject/demoapp1/models.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+from django.core.validators import validate_email, MaxValueValidator
+
+
+# abstract model
+class BaseModel(models.Model):
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def get_all(cls):
+        return cls.objects.all()
+
+    @classmethod
+    def filter_objects(cls, **kwargs):
+        return cls.objects.filter(**kwargs)
+
+    @classmethod
+    def get_first(cls, **kwargs):
+        return cls.objects.filter(**kwargs).first()
+
+
+class CustomUser(AbstractUser, BaseModel):
+    user_id = models.AutoField(primary_key=True,)
+    email = models.EmailField(max_length=45, unique=True,
+                              validators=[validate_email])
+    phone_number = models.CharField(max_length=100, unique=True)
+    username = models.CharField(max_length=100, blank=True, null=True)
+    USERNAME_FIELD = 'email'  # unique id
+    REQUIRED_FIELDS = ['username']
+
```

Comment:
- Signup (Add User)
- Login (Verify User)
- Check if user already registered for an email
- Can we get a list of Borrowed Books? Like this: `user.borrowed_books`

Reply:
used `related_name`